src/wp-includes/media.php

```diff
@@ -1965,17 +1972,70 @@ function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
 	 */
 	$optimization_attrs = wp_get_loading_optimization_attributes(
 		'img',
-		array(
-			'width'         => $width,
-			'height'        => $height,
-			'loading'       => $loading_val,
-			'fetchpriority' => $fetchpriority_val,
-		),
+		$args,
 		$context
 	);
 
-	// Images should have source and dimension attributes for the loading optimization attributes to be added.
-	if ( ! str_contains( $image, ' src="' ) || ! str_contains( $image, ' width="' ) || ! str_contains( $image, ' height="' ) ) {
+	// Images should have source for the decoding and the loading optimization attributes to be added.
+	if ( ! str_contains( $image, ' src="' ) ) {
+		return $image;
+	}
+
+	if ( empty( $decoding_val ) ) {
+		/**
+		 * Filters the `decoding` attribute value to add to an image. Default `async`.
+		 *
+		 * Returning a falsey value will omit the attribute.
+		 *
+		 * @since 6.1.0
+		 *
+		 * @param string|false|null $value      The `decoding` attribute value. Returning a falsey value
+		 *                                      will result in the attribute being omitted for the image.
+		 *                                      Otherwise, it may be: 'async' (default), 'sync', or 'auto'.
+		 * @param string            $image      The HTML `img` tag to be filtered.
+		 * @param string            $context    Additional context about how the function was called
+		 *                                      or where the img tag is.
+		 */
+		$filtered_decoding_attr = apply_filters( 'wp_img_tag_add_decoding_attr', 'async', $image, $context );
```

Comment:
The filter should respect the value returned by `wp_get_loading_optimization_attributes()`, as that is now the source of truth function, so it shouldn't be hard-coded to `async` here. Realistically, in 99% of cases the result will still be `async`, but the decision should come from the new source of truth. Otherwise it may accidentally override the value from that function even when the filter isn't actually used.
```suggestion
		$filtered_decoding_attr = apply_filters(
			'wp_img_tag_add_decoding_attr',
			isset( $optimization_attrs['decoding'] ) ? $optimization_attrs['decoding'] : false,
			$image,
			$context
		);
```

Reply:
@felixarntz 

The reason why I kept the `async` here is because that's the current [default](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/media.php#L2087) value. 

So, will something like this keep the retrocompatibility?
```suggestion
		$filtered_decoding_attr = apply_filters(
			'wp_img_tag_add_decoding_attr',
			isset( $optimization_attrs['decoding'] ) ? $optimization_attrs['decoding'] : 'async',
			$image,
			$context
		);
```
wdyt?

Reply:
@pereirinha I think that actually doesn't make a difference by default. With the new change, `wp_get_loading_optimization_attributes()` should always return `decoding="async"` for images that don't have the attribute yet, so it would always be set here.

I think the reason we should avoid using `async` as default here is that, per https://core.trac.wordpress.org/ticket/58893, developers could modify what `wp_get_loading_optimization_attributes()` returns, and then that should be reflected here as well. It wouldn't work if we put `async` as a default.